src/utils/commitFile.ts

```diff
@@ -54,30 +45,68 @@ export const commitFileMac = async (
 
   try {
     gitCommands =
-      isWindows === true
-        ? (windowsCommands(owner, repo, branch, fileName) as commands)
-        : isWindows === false && isCodespace === false
-        ? (macCommands(owner, repo, branch, fileName) as commands)
-        : (codespacesCommands(owner, repo, branch, fileName) as commands);
+      root === "workspaces"
+        ? (codespacesCommands(owner, repo, branch, fileName) as commands)
+        : (generalCommands(owner, repo, branch, fileName) as commands);
     inform(gitCommands);
   } catch (err) {
     error(err);
     throw err;
   }
 
   for (index = 0; index < gitCommands.length; index++) {
-    const { stdout, stderr } = await execFile(
-      gitCommands[index].command,
-      gitCommands[index].args,
-      {
-        cwd: gitCommands[index].cwd,
-      }
+    inform(
+      [
+        "Executing: ",
+        gitCommands[index].command,
+        gitCommands[index].args,
+        "in",
+        gitCommands[index].cwd,
+      ].join(" ")
     );
-    if (stderr) {
-      error(stderr);
+    // Adding try/catch so we can whitelist
+    try {
+      const { stdout, stderr } = await execFile(
+        gitCommands[index].command,
+        gitCommands[index].args,
+        {
+          cwd: gitCommands[index].cwd,
+          shell: true,
+        }
+      );
+      if (stderr) {
+        error(stderr);
+      }
+      inform(stdout);
+      await delay(1000);
+    } catch (err: any) {
+      inform(`Whitelist returns: ${whiteListed(err.message)}`);
+      if (!whiteListed(err.message)) {
+        throw err;
+      }
     }
-    inform(stdout);
-    await delay(1000);
   }
   return { status: 200, message: "success" };
 };
+
+/**
+ *
+ * @param errorMsg    The string error message captured
+ * @returns           A boolean determined by the existance or lack of a
+ *                    whitelist match.
+ */
+function whiteListed(errorMsg: string): boolean {
```

Comment:
What is the purpose of this function 🤔 I see you are passing in an error message and then seeing if it contains anything in the `whiteList` array 🤔 why would you only be throwing an error then 🤔

Reply:
This function is to bypass known/accepted errors. Errors like: "The system cannot find the file specified" which occurred whentrying to blanket purge the destdir as the 1st command. "cannot make this folder because it already exist" which occurred when the `.github/workflows` folder already existed. 

If I didn't delete destdir 1st after a failed run I'd get an error saying the folder already existed on the next command mkdir. It was REALLY annoying to run the program, error out, then go find the folder it made delete it, then run it again. So, I just put a rmdir as the 1st command and ignored the error.

Windows doesn't always ignore stuff like that as Linux/MacOS does.

Reply:
sounds good to me 🙇 I will keep this in 🙇